src/samplereader/FragmentedSampleReader.cpp

```diff
@@ -319,17 +319,18 @@ AP4_Result CFragmentedSampleReader::ProcessMoof(AP4_ContainerAtom* moof,
                                                 AP4_Position mdat_payload_offset,
                                                 AP4_UI64 mdat_payload_size)
 {
-  // For prefixed initialization (usually ISM) we don't yet know the
-  // proper track id, let's find it now
-  if (m_track->GetId() == TRACKID_UNKNOWN)
   {
     AP4_MovieFragment fragment =
         AP4_MovieFragment(AP4_DYNAMIC_CAST(AP4_ContainerAtom, moof->Clone()));
     AP4_Array<AP4_UI32> ids;
     fragment.GetTrackIds(ids);
-    if (ids.ItemCount() == 1)
+    // If track ID changed due to representation switch,
+    // update the tracker
```

Comment:
if this is a workaround and there is the need to find a better solution
i think would better change this with some TODO comments

something like this, feel free to improve it
```suggestion
  //! @todo: To implement support for multiple tracks within a segment.
  //! @todo: When we play an adaptive stream and the stream quality change during playback the track ID change and we force update it, but may be need to find an appropriate solution to update the track.
```